x-pack/plugins/uptime/public/components/fleet_package/certs_field.test.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { fireEvent, waitFor } from '@testing-library/react';
+import { render } from '../../lib/helper/rtl_helpers';
+import { CertsField, SSLRole, FieldKey, VerificationMode, FieldConfig } from './certs_field';
+
+// ensures that fields appropriately match to their label
+jest.mock('@elastic/eui/lib/services/accessibility/html_id_generator', () => ({
+  htmlIdGenerator: () => () => `id-${Math.random()}`,
+}));
+
+describe('<CertsField />', () => {
+  const onChange = jest.fn();
+  const WrappedComponent = ({
+    sslRole = SSLRole.CLIENT,
+    fieldConfig,
+  }: {
+    sslRole?: SSLRole;
+    fieldConfig?: FieldConfig;
+  }) => {
+    return <CertsField onChange={onChange} sslRole={sslRole} fieldConfig={fieldConfig} />;
+  };
+  it('renders CertsField', () => {
+    const { getByLabelText, getByText } = render(<WrappedComponent />);
+
+    expect(getByText('Certificate settings')).toBeInTheDocument();
+    expect(getByLabelText('Client certificate')).toBeInTheDocument();
+    expect(getByLabelText('Client key')).toBeInTheDocument();
+    expect(getByLabelText('Certificate authorities')).toBeInTheDocument();
+    expect(getByLabelText('Verification mode')).toBeInTheDocument();
+  });
+
+  it('handles role', () => {
+    const { getByLabelText, rerender } = render(<WrappedComponent sslRole={SSLRole.SERVER} />);
+
+    expect(getByLabelText('Server certificate')).toBeInTheDocument();
+    expect(getByLabelText('Server key')).toBeInTheDocument();
+
+    rerender(<WrappedComponent sslRole={SSLRole.CLIENT} />);
+  });
+
+  it('updates fields and calls onChange', async () => {
+    const { getByLabelText } = render(<WrappedComponent />);
+
+    const clientCertificate = getByLabelText('Client certificate') as HTMLInputElement;
```

Comment:
i am surprised that you need this `as HTMLInputElement` here

Reply:
Yes, I need to in order to access `clientCertificate.value` without typescript complaining. It is surprising though. You would think getByLabelText would return HTMLInputElement type.